File: app/src/main/java/openfoodfacts/github/scrachx/openfood/dagger/component/ActivityComponent.java
package org.openfoodfacts.scanner.dagger.component;

import dagger.Subcomponent;
import org.openfoodfacts.scanner.dagger.ActivityScope;
import org.openfoodfacts.scanner.dagger.module.ActivityModule;
import org.openfoodfacts.scanner.views.BaseActivity;

@Subcomponent(modules = {ActivityModule.class})
@ActivityScope
public interface ActivityComponent {

    FragmentComponent plusFragmentComponent();

    void inject(BaseActivity baseActivity);
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/dagger/component/ActivityComponent.kt
package openfoodfacts.github.scrachx.openfood.dagger.component

import dagger.Subcomponent
import openfoodfacts.github.scrachx.openfood.dagger.ActivityScope
import openfoodfacts.github.scrachx.openfood.dagger.module.ActivityModule
import openfoodfacts.github.scrachx.openfood.features.shared.BaseActivity

@Subcomponent(modules = [ActivityModule::class])
@ActivityScope
interface ActivityComponent {
    fun plusFragmentComponent(): FragmentComponent?
    fun inject(baseActivity: BaseActivity?)
}